### ISSUE TITLE
Would you mind to remove these packages from go.mod and go.sum

### ISSUE THREAD

**Issue by Blank-Xu** [jmoiron/sqlx#642]
Would you mind to remove these packages from go.mod and go.sum?

`
	github.com/go-sql-driver/mysql v1.4.0
	github.com/lib/pq v1.0.0
	github.com/mattn/go-sqlite3 v1.9.0
`

These three packages just used in test files and does not necessary to include in sqlx package.

**Comment by Indribell** [comment]
Indeed... We also looked at why so many dependencies are being included into our project, when we use the pure ( none CGO ) Go version of sqlite ( https://gitlab.com/cznic/sqlite ). Only to see that its all coming from sqlx.

**Comment by arp242** [comment]
There isn't really any good solution for this; removing them means they will get re-added on `go test`, and you do want some versioning for test dependencies too. Modules also don't really have an option to list "test dependencies".

**Comment by jwendel** [comment]
I'm wondering if the [go.work](https://go.dev/doc/tutorial/workspaces) file would be the way to go here.  There could be the main library in one path, then all the tests in a separate path with its own go.mod file.

**Comment by krwu** [comment]
Users of sqlx can use this library with any database driver, however, the tests force them to install unnecessary MySQL, PostgreSQL or SQLite driver, and the `mattn/go-sqlite3` cause users have to enable CGO.

**Comment by mrj0** [comment]
Hm so I build without CGO and the test dependencies are correctly excluded from my resulting bundle. Is that not working correctly for you?

Moving to a tests package could work but many of the sql testing functions use module private functions and would require some rework.
